apps/studio/components/ui/AdvisorPanel/AdvisorPanel.tsx

```diff
@@ -0,0 +1,331 @@
+    case 'WARN':
+      return 'warning'
+    default:
+      return 'info'
+  }
+}
+
+export const AdvisorPanel = () => {
+  const {
+    activeTab,
+    severityFilters,
+    selectedItemId,
+    setActiveTab,
+    setSeverityFilters,
+    clearSeverityFilters,
+    setSelectedItemId,
+  } = useAdvisorStateSnapshot()
+  const { data: project } = useSelectedProjectQuery()
+  const { activeSidebar, closeSidebar } = useSidebarManagerSnapshot()
+
+  const isSidebarOpen = activeSidebar?.id === SIDEBAR_KEYS.ADVISOR_PANEL
+
+  const { data: lintData, isLoading: isLintsLoading } = useProjectLintsQuery(
+    { projectRef: project?.ref },
+    { enabled: isSidebarOpen && !!project?.ref }
+  )
+
+  const lintItems = useMemo<AdvisorItem[]>(() => {
+    if (!lintData) return []
+
+    return lintData
+      .map((lint): AdvisorItem | null => {
+        const categories = lint.categories || []
+        const tab =
+          categories.includes('SECURITY') && !categories.includes('PERFORMANCE')
+            ? ('security' as const)
+            : categories.includes('PERFORMANCE') && !categories.includes('SECURITY')
+              ? ('performance' as const)
+              : categories.includes('SECURITY')
+                ? ('security' as const)
+                : categories.includes('PERFORMANCE')
+                  ? ('performance' as const)
+                  : undefined
+
+        if (!tab) return null
+
+        return {
+          id: lint.cache_key,
+          title: lint.detail,
+          severity: lintLevelToSeverity(lint.level),
+          createdAt: undefined,
+          tab,
+          source: 'lint' as const,
+          original: lint,
+        }
+      })
+      .filter((item): item is AdvisorItem => item !== null)
+  }, [lintData])
+
+  const combinedItems = useMemo<AdvisorItem[]>(() => {
+    const all = [...lintItems]
+
+    return all.sort((a, b) => {
+      const severityDiff = severityOrder[a.severity] - severityOrder[b.severity]
+      if (severityDiff !== 0) return severityDiff
+
+      const createdDiff = (b.createdAt ?? 0) - (a.createdAt ?? 0)
+      if (createdDiff !== 0) return createdDiff
+
+      return a.title.localeCompare(b.title)
+    })
+  }, [lintItems])
+
+  const filteredItems = useMemo<AdvisorItem[]>(() => {
+    return combinedItems.filter((item) => {
+      if (severityFilters.length > 0 && !severityFilters.includes(item.severity)) {
+        return false
+      }
+
+      if (activeTab === 'all') return true
+
+      return item.tab === activeTab
+    })
+  }, [combinedItems, severityFilters, activeTab])
+
+  const itemsFilteredByTabOnly = useMemo<AdvisorItem[]>(() => {
+    return combinedItems.filter((item) => {
+      if (activeTab === 'all') return true
+      return item.tab === activeTab
+    })
+  }, [combinedItems, activeTab])
+
+  const hiddenItemsCount = itemsFilteredByTabOnly.length - filteredItems.length
+
+  const selectedItem = filteredItems.find((item) => item.id === selectedItemId)
```

Comment:
It's possible for this to get into an odd state, where you click something that _should_ open the details view, but doesn't, because the already-active filters filter it out